middlewares/user/getTargetUserIdByPMChannelId.js

```diff
@@ -0,0 +1,57 @@
+const {StreamChat} = require('stream-chat');
+const Getstream = require('../../vendor/getstream');
+
+/**
+ *
+ * @param {import('express').Request} req
+ * @param {import('express').Response} res
+ * @param {import('express').NextFunction} next
+ * @returns
+ */
+const getTargetUserIdByPMChannelIdMiddleware = async (req, res, next) => {
+  const {userId} = req || {};
+  const {channelId} = req?.query || {};
+
+  if (!channelId) {
+    return res.status(400).json({
+      code: 400,
+      success: false,
+      message: 'Channel ID is required'
+    });
+  }
+
+  const chatClient = new StreamChat(process.env.API_KEY, process.env.SECRET);
+  let channelMembers;
+  try {
+    let {users} = await Getstream.chat.getTargetUserIdByChannelId(chatClient, userId, channelId);
+    channelMembers = users;
+  } catch (e) {
+    console.error('Failed to get channel members', e);
+    return res.status(500).json({
+      code: 500,
+      success: false,
+      message: 'Internal server error'
+    });
+  }
+
+  if (!channelMembers || channelMembers?.length !== 1)
+    return res.status(400).json({
+      code: 400,
+      success: false,
+      message: 'Not a valid PM channel'
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The condition checks if `channelMembers` is not defined or if its length is not equal to 1. However, it does not handle the case where `channelMembers` is not an array which could lead to a TypeError.
```diff
-  if (!channelMembers || channelMembers?.length !== 1)
+  if (!Array.isArray(channelMembers) || channelMembers.length !== 1)
     return res.status(400).json({
       code: 400,
       success: false,
       message: 'Not a valid PM channel'
     });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->